fili-core/src/test/groovy/com/yahoo/bard/webservice/web/endpoints/DruidQueryIdSpec.groovy

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.endpoints
+
+import com.yahoo.bard.webservice.data.dimension.BardDimensionField
+import com.yahoo.bard.webservice.data.dimension.DimensionDictionary
+import com.yahoo.bard.webservice.util.JsonSlurper
+import com.yahoo.bard.webservice.util.JsonSortStrategy
+import com.yahoo.bard.webservice.web.filters.BardLoggingFilter
+
+import javax.ws.rs.core.MultivaluedHashMap
+
+
+class DruidQueryIdSpec extends BaseDataServletComponentSpec {
+    def prefixId = "abcdef"
+
+    @Override
+    def setup() {
+        DimensionDictionary dimensionStore = jtb.configurationLoader.dimensionDictionary
+        dimensionStore.findByApiName("color").with {
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "Foo", "FooDesc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "Bar", "BarDesc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "Baz", "BazDesc"))
+        }
+    }
+
+    @Override
+    Class<?>[] getResourceClasses() {
+        [ DataServlet.class, BardLoggingFilter.class ]
+    }
+
+    @Override
+    String getTarget() {
+        return "data/shapes/week/color"
+    }
+
+    @Override
+    Map<String, List<String>> getQueryParams() {
+        [
+                "metrics": ["width","depth"],
+                "dateTime": ["2014-06-02/2014-06-30"],
+                "sort": ["width|desc","depth|asc"]
+        ]
+    }
+
+    @Override
+    boolean compareResult(String result, String expectedResult, JsonSortStrategy sortStrategy = JsonSortStrategy.SORT_MAPS) {
+        JsonSlurper slurper = new JsonSlurper(sortStrategy)
+        def parsedJson = slurper.parseText(result)
+        if (parsedJson.context != null) {
+            testedDruidQuery = true
+            return parsedJson.context.queryId.startsWith(prefixId)
+        }
```

Comment:
This didn't actually work. This fails in the case where we want this function to act as a noop.